### DISCUSSION TITLE
Add warning to 2.x versions on install to upgrade to Meteor 3

### DISCUSSION THREAD

**Discussion by jankapunkt** [meteor/meteor#12882] [Feature Requests]
I found again, that people looked into Meteor by download / install and running to realize it runs on Node 14. For someone who does not know about the whole Async/Fibers issue it's a red flag.

I therefore think the install/update should warn the user with an understandable message:

```shell
This Meteor 2.x version is running on a deprecated Node.js version (14). You are advised to install/upgrade to the latest 3.0 release to be on the latest Node.js version. For more information, please consult this guide: https://guide.meteor.com/prepare-meteor-3.0
```

This is to catch the attention of all the people who were not into this whole Fibers migration topic.

**Comment by DanielDornhardt** [comment]
I think it's a good idea!

Especially for people starting with a new project it'd be great if they would be led to some docs on how to start with Meteor 3.x.

**Comment by DanielDornhardt** [comment]
Problem being of course, Meteor 3 is still alpha, but I think it's _almost_ good to go for starting with?

**Reply by jankapunkt** [reply]
Beta closing in soon

**Reply by StorytellerCZ** [reply]
Agreed, once we are in Beta or RC, then I think it makes sense. This is a must after 3.0 is released.

**Comment by harryadel** [comment]
Yeah, I think adding it will definitely help.

**Comment by Grubba27** [comment]
When running `curl https://install.meteor.com/ | sh` or when installing via `meteor npm i -g`, correct?

**Reply by jankapunkt** [reply]
Yes, exactly and I think this should be included asap, since there are still people that get into this situation right now

**Reply by fredmaiaarantes** [reply]
Perhaps we should adjust the message to mention Meteor 3.0 but do not recommend installing it yet. When beta is out, we could change the message. So I think it could be:

```shell
This Meteor 2.x version is running on a deprecated Node.js version (14). Meteor 3.0 is in progress, and it will run on the latest Node.js version. For more information, please consult this migration guide: https://guide.meteor.com/3.0-migration.
```

The page mentioned is the one I proposed to be created on this [PR](https://github.com/meteor/meteor/pull/12883).

**Comment by zodern** [comment]
Something else that might help is currently the npm installer requires Node 14. It is probably a small change to make it work with newer node versions (making the global install check work with newer npm versions, or simply removing it). This wouldn't fix Meteor itself using node 14, but at least you wouldn't have to install node 14 to install Meteor (Meteor comes with its own copy of node and npm and doesn't use the global version).

**Reply by jankapunkt** [reply]
It is this one here, right? https://github.com/meteor/meteor/tree/devel/npm-packages/meteor-installer

**Reply by zodern** [reply]
Yes. It restricts the node and npm version at https://github.com/meteor/meteor/blob/660025db826153ce6b0ddb2f6723381fa861e854/npm-packages/meteor-installer/package.json#L26-L27 and https://github.com/meteor/meteor/blob/660025db826153ce6b0ddb2f6723381fa861e854/npm-packages/meteor-installer/install.js#L38-L46

When I had tried it with Node 16 a while ago, it seemed the only problem was [this check](https://github.com/meteor/meteor/blob/660025db826153ce6b0ddb2f6723381fa861e854/npm-packages/meteor-installer/install.js#L48-L59).

**Reply by jankapunkt** [reply]
@zodern but what's the issue exactly with the global npm context check?
